numpyro/infer/mcmc.py

```diff
@@ -499,17 +497,12 @@ def run(self, rng_key, *args, extra_fields=(), init_params=None, **kwargs):
             states_flat, last_state = partial_map_fn(map_args)
             states = tree_map(lambda x: x[jnp.newaxis, ...], states_flat)
         else:
-            if chain_method == 'sequential':
-                if self.progress_bar:
-                    states, last_state = _laxmap(partial_map_fn, map_args)
-                else:
-                    states, last_state = lax.map(partial_map_fn, map_args)
-            elif chain_method == 'parallel':
+            if self.chain_method == 'sequential':
+                states, last_state = _laxmap(partial_map_fn, map_args)
```

Comment:
Using `lax.map` here is a bit faster but expensive in term of memory requirement so I switched to for loop. See the benchmark
```python
%%time
%%memit

import jax

import numpyro; numpyro.set_platform("gpu")
import numpyro.distributions as dist
from numpyro.infer import MCMC, NUTS

def model():
    numpyro.sample("x", dist.Normal(0, 1).expand([100]))

mcmc = MCMC(NUTS(model), 1000, num_samples=10000, progress_bar=False, num_chains=8)
mcmc.run(jax.random.PRNGKey(0))
samples = mcmc.get_samples(group_by_chain=True)["x"]
print(samples.shape)
```
#### using lax.map
```
(8, 10000, 100)
peak memory: 5462.81 MiB, increment: 5415.39 MiB
CPU times: user 9min 23s, sys: 6.47 s, total: 9min 29s
Wall time: 9min 25s
```
while using for loop `_laxmap`:
```
(8, 10000, 100)
peak memory: 1795.53 MiB, increment: 1748.04 MiB
CPU times: user 9min 41s, sys: 5.9 s, total: 9min 47s
Wall time: 9min 44s
```

Reply:
This seems more than acceptable for the memory savings!